PowerShell Scanners/ProduKey/Setup.ps1

```diff
@@ -0,0 +1,10 @@
+# Disable progress bars.
+$ProgressPreference = 'SilentlyContinue'
+
+Remove-Item -Path '.\ProduKey.exe' -ErrorAction 'SilentlyContinue'
+
+[System.Net.WebClient]::new().DownloadFile('https://www.nirsoft.net/utils/produkey.zip', "$PWD\produkey.zip")
+
+Expand-Archive -Path '.\produkey.zip' -DestinationPath .
```

Comment:
Because the of scan profile.xml using a absolute path, I think that this script should use more absolute paths to make sure that the ProduKey.exe gets put in the right place so that it can be used with the scanner. If the script is called from another path it will save the file in the current directory instead of with the scanner.

Reply:
A possible way to do this is like so:

```pwsh
$a = [System.IO.Path]::GetTempFileName()
$d = [System.IO.Path]::GetDirectoryName($a)
[System.Net.WebClient]::new().DownloadFile('https://www.nirsoft.net/utils/produkey.zip', $a)
Expand-Archive -Path $a -DestinationPath $d
```

Additionally Windows Defender really doesn't like it when this series of commands are run, I think it's too close to malware as for the download and extraction I got the respective false positives below.

https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?name=Trojan%3aWin32%2fSkeeyah&threatid=2147708902

https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?name=Trojan%3aWin32%2fTnega!ml&threatid=2147763770

Reply:
You can also make the clean up done on line 10 dynamic by using the -PassThru parameter on Expand-Archive. It will return file objects of the files that were successfully extracted.

Reply:
> Because the of scan profile.xml using a absolute path, I think that this script should use more absolute paths to make sure that the ProduKey.exe gets put in the right place so that it can be used with the scanner. If the script is called from another path it will save the file in the current directory instead of with the scanner.

I think I found a more elegant solution. Let me know what you think.

Reply:
> Additionally Windows Defender really doesn't like it when this series of commands are run, I think it's too close to malware as for the download and extraction I got the respective false positives below.

Yep, that's why I made a note in this scanner's README.

Reply:
> You can also make the clean up done on line 10 dynamic by using the -PassThru parameter on Expand-Archive. It will return file objects of the files that were successfully extracted.

That would be neat, but they added that parameter in PS Core. https://github.com/PowerShell/Microsoft.PowerShell.Archive/pull/18